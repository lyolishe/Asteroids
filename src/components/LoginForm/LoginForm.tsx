import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import {
  Form, Input, Button, Row, Col, Card,
} from 'antd';
import { useForm } from 'antd/es/form/Form';
import { useHistory } from 'react-router';
import AuthApi from '../../api/AuthApi';
import { SignUser } from '../../types/types';
import { openNotificationWithIcon } from '../../helpers/NotificationHelper';

const layout = {
  labelCol: { span: 8 },
  wrapperCol: { span: 16 },
};
const tailLayout = {
  wrapperCol: { offset: 8, span: 16 },
};

const LoginForm = (): JSX.Element => {
  const [form] = useForm();
  const history = useHistory();

  const onFinish = (values: SignUser) => {
    AuthApi.signIn(values)
      .then(() => {
        openNotificationWithIcon('success', 'Успех', '');
        history.push('/');
      })
      .catch((err) => {
        openNotificationWithIcon('error', 'Ошибка', err.response.data.reason);
      });
  };

  const toRegistration = () => {
    history.push('/registration');
  };

  useEffect(() => {
    AuthApi.getUser()
      .then(() => {
        history.push('/');
      })
      .catch(() => {
      });
  });

  return (
    <Form
      {...layout}
      name="basic"
      onFinish={onFinish}
      layout="vertical"
      hideRequiredMark
      form={form}
    >
      <Row>
        <Col span={12} offset={6}>
          <Card title="Авторизация">
            <Form.Item
              label="Логин"
              name="login"
              rules={[{ required: true, message: 'Заполните поле!' }]}
            >
              <Input />
            </Form.Item>

            <Form.Item
              label="Пароль"
              name="password"
              rules={[{ required: true, message: 'Заполните поле!' }]}
            >
              <Input.Password />
            </Form.Item>

            <Form.Item {...tailLayout}>
              <Button type="primary" htmlType="submit">
                Войти
              </Button>
            </Form.Item>

            <Form.Item {...tailLayout}>
              <Link to="/register">
                <Button htmlType="button" onClick={toRegistration}>
                  Нет аккаунта?
                </Button>
              </Link>
            </Form.Item>
          </Card>
        </Col>
      </Row>
    </Form>
  );
};

export default LoginForm;
