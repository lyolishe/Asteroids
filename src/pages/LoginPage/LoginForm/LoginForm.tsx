import React from 'react';
import { Link } from 'react-router-dom';
import {
  Form, Input, Button, Row, Col, Card,
} from 'antd';
import { useForm } from 'antd/es/form/Form';

const layout = {
  labelCol: { span: 8 },
  wrapperCol: { span: 16 },
};
const tailLayout = {
  wrapperCol: { offset: 8, span: 16 },
};

const LoginForm = (): JSX.Element => {
  const [form] = useForm();

  const onFinish = (values: any) => {
    console.log('Success:', values);
  };

  const onFinishFailed = (errorInfo: any) => {
    console.log('Failed:', errorInfo);
  };

  return (
    <Form
      {...layout}
      name="basic"
      onFinish={onFinish}
      onFinishFailed={onFinishFailed}
      layout="vertical"
      hideRequiredMark
      form={form}
    >
      <Row>
        <Col span={12} offset={6}>
          <Card title="Авторизация">
            <Form.Item
              label="Логин"
              name="login"
              rules={[{ required: true, message: 'Заполните поле!' }]}
            >
              <Input />
            </Form.Item>

            <Form.Item
              label="Пароль"
              name="password"
              rules={[{ required: true, message: 'Заполните поле!' }]}
            >
              <Input.Password />
            </Form.Item>

            <Form.Item {...tailLayout}>
              <Button type="primary" htmlType="submit">
                Войти
              </Button>
            </Form.Item>

            <Form.Item {...tailLayout}>
              <Link to="/register">
                <Button htmlType="button">
                  Нет аккаунта?
                </Button>
              </Link>
            </Form.Item>
          </Card>
        </Col>
      </Row>
    </Form>
  );
};

export default LoginForm;
