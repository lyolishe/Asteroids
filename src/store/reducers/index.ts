import { combineReducers } from 'redux';
import leaderboardReducer from './leaderboard';
import authReducer from './auth';

const reducers = combineReducers({
  leaderboard: leaderboardReducer,
  auth: authReducer,
});

export default reducers;
