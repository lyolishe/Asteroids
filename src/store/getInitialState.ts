import { RouterState } from 'connected-react-router';
import { State } from '../types';
import { initialState as auth } from './reducers/auth';
import { initialState as leaderboard } from './reducers/leaderboard';

const getInitialState = (pathname = '/'): State => ({
  auth,
  leaderboard,
  router: {
    location: {
      pathname, search: '', hash: '', key: '',
    },
    action: 'POP',
  } as RouterState,
});

export default getInitialState;
