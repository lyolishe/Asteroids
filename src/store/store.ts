import createSagaMiddleware, { END } from 'redux-saga';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import { routerMiddleware, connectRouter } from 'connected-react-router';
import { createBrowserHistory, createMemoryHistory } from 'history';
import reducers from './reducers';
import { isServer } from '../helpers/isServer';

import watchGotLeaderboard from './sagas/leaderboard';
import watchLogin from './sagas/auth';

import { AppStore, State } from '../types';

// eslint-disable-next-line import/prefer-default-export
export function configureStore(initialState: any, url = '/') {
  const history = isServer
    ? createMemoryHistory({ initialEntries: [url] })
    : createBrowserHistory();

  const sagaMiddleware = createSagaMiddleware();
  const middlewares = [routerMiddleware(history), sagaMiddleware];

  const store = createStore(
    combineReducers<State>({
      ...reducers,
      router: connectRouter(history),
    }),
    initialState,
    applyMiddleware(...middlewares),
  ) as AppStore;

  store.runSaga = sagaMiddleware.run;
  store.close = () => store.dispatch(END);

  if (!isServer) {
    sagaMiddleware.run(watchLogin);
    sagaMiddleware.run(watchGotLeaderboard);
  }

  return { store, history };
}
