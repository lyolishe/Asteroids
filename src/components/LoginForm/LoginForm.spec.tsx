import React from 'react';
import { Provider as ReduxProvider } from 'react-redux';
import { render, RenderResult } from '@testing-library/react';
import { SnackbarProvider } from 'notistack';

import { configureStore } from '../../store/store';
import getInitialState from '../../store/getInitialState';

import LoginForm from './LoginForm';

let documentBody: RenderResult;

describe('<LoginForm />', () => {
  beforeEach(() => {
    const location = '/login';
    const { store } = configureStore(getInitialState(location), location);
    documentBody = render(
      <ReduxProvider store={store}>
        <SnackbarProvider>
          <LoginForm />
        </SnackbarProvider>
      </ReduxProvider>,
    );
  });

  it('matches snapshot', () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
