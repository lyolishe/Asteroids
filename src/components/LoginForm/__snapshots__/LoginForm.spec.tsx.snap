// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> matches snapshot 1`] = `
<body>
  <div>
    <form
      novalidate=""
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
          data-shrink="true"
        >
          Логин
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="login"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
            >
              <span>
                Логин
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
        >
          Пароль
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="current-password"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="password"
            required=""
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-7"
            >
              <span>
                Пароль
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Войти
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Войти через Яндекс
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="/register"
          >
            Нет аккаунта?
          </a>
        </div>
      </div>
    </form>
  </div>
</body>
`;
